### ISSUE TITLE
npm install fails on Windows

### ISSUE THREAD

**Issue by coryhouse** [rackt/react-router#1577]
Installs fine on my Mac, but installing on Windows via Git Bash, npm install fails. Note, I have to provide the msvs_version flag due to a separate issue with node-gyp (http://stackoverflow.com/questions/14278417/cannot-install-node-modules-that-require-compilation-on-windows-7-x64-vs2012/16854333#16854333)

$ npm install --msvs_version=2013
npm WARN package.json babel-core@5.7.4 No README data
npm WARN package.json events@1.0.2 events is also the name of a node core module.
npm WARN package.json karma-mocha@0.1.10 No README data
npm WARN package.json rf-changelog@0.4.0 No repository field.

> react-router@1.0.0-beta3 prepublish c:\Users\chouse\Desktop\react-router
> npm run build

> react-router@1.0.0-beta3 build c:\Users\chouse\Desktop\react-router
> scripts/build.sh

'scripts' is not recognized as an internal or external command,
operable program or batch file.

npm ERR! react-router@1.0.0-beta3 build: `scripts/build.sh`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the react-router@1.0.0-beta3 build script.
npm ERR! This is most likely a problem with the react-router package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     scripts/build.sh
npm ERR! You can get their info via:
npm ERR!     npm owner ls react-router
npm ERR! There is likely additional logging output above.
npm ERR! System Windows_NT 6.1.7601
npm ERR! command "c:\\Program Files (x86)\\nodejs\\\\node.exe" "c:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "
build"
npm ERR! cwd c:\Users\chouse\Desktop\react-router
npm ERR! node -v v0.10.39
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     c:\Users\chouse\Desktop\react-router\npm-debug.log
npm ERR! not ok code 0

npm ERR! react-router@1.0.0-beta3 prepublish: `npm run build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the react-router@1.0.0-beta3 prepublish script.
npm ERR! This is most likely a problem with the react-router package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build
npm ERR! You can get their info via:
npm ERR!     npm owner ls react-router
npm ERR! There is likely additional logging output above.
npm ERR! System Windows_NT 6.1.7601
npm ERR! command "c:\\Program Files (x86)\\nodejs\\node.exe" "c:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
 "--msvs_version=2013"
npm ERR! cwd c:\Users\chouse\Desktop\react-router
npm ERR! node -v v0.10.39
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     c:\Users\chouse\Desktop\react-router\npm-debug.log
npm ERR! not ok code 0

**Comment by coryhouse** [comment]
And here's the content of npm-debug.log:

chouse@VIN7D100587 /c/Users/chouse/Desktop/react-router (master)
$ cat npm-debug.log
0 info it worked if it ends with ok
1 verbose cli [ 'c:\\Program Files (x86)\\nodejs\\node.exe',
1 verbose cli   'c:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'install',
1 verbose cli   '--msvs_version=2013' ]
2 info using npm@1.4.28
3 info using node@v0.10.39
4 verbose readDependencies using package.json deps
5 verbose install where, deps [ 'c:\\Users\\chouse\\Desktop\\react-router',
5 verbose install   [ 'invariant',
5 verbose install     'keymirror',
5 verbose install     'qs',
5 verbose install     'warning',
5 verbose install     'babel',
5 verbose install     'babel-core',
5 verbose install     'babel-eslint',
5 verbose install     'babel-loader',
5 verbose install     'babel-plugin-object-assign',
5 verbose install     'bundle-loader',
5 verbose install     'eslint',
5 verbose install     'events',
5 verbose install     'expect',
5 verbose install     'karma',
5 verbose install     'karma-chrome-launcher',
5 verbose install     'karma-cli',
5 verbose install     'karma-firefox-launcher',
5 verbose install     'karma-mocha',
5 verbose install     'karma-sourcemap-loader',
5 verbose install     'karma-webpack',
5 verbose install     'marked',
5 verbose install     'mocha',
5 verbose install     'pygmentize-bundled',
5 verbose install     'react',
5 verbose install     'rf-changelog',
5 verbose install     'rx',
5 verbose install     'slash',
5 verbose install     'webpack',
5 verbose install     'webpack-dev-server' ] ]
6 info preinstall react-router@1.0.0-beta3
7 warn package.json babel-core@5.7.4 No README data
8 warn package.json events@1.0.2 events is also the name of a node core module.
9 warn package.json karma-mocha@0.1.10 No README data
10 warn package.json rf-changelog@0.4.0 No repository field.
11 verbose readDependencies using package.json deps
12 verbose already installed skipping invariant@^2.0.0 c:\Users\chouse\Desktop\react-router
13 verbose already installed skipping keymirror@^0.1.1 c:\Users\chouse\Desktop\react-router
14 verbose already installed skipping qs@2.4.1 c:\Users\chouse\Desktop\react-router
15 verbose already installed skipping warning@^2.0.0 c:\Users\chouse\Desktop\react-router
16 verbose already installed skipping babel@^5.4.7 c:\Users\chouse\Desktop\react-router
17 verbose already installed skipping babel-core@^5.4.7 c:\Users\chouse\Desktop\react-router
18 verbose already installed skipping babel-eslint@^3.1.23 c:\Users\chouse\Desktop\react-router
19 verbose already installed skipping babel-loader@^5.0.0 c:\Users\chouse\Desktop\react-router
20 verbose already installed skipping babel-plugin-object-assign@^1.2.0 c:\Users\chouse\Desktop\react-router
21 verbose already installed skipping bundle-loader@^0.5.2 c:\Users\chouse\Desktop\react-router
22 verbose already installed skipping eslint@^1.0.0-rc-1 c:\Users\chouse\Desktop\react-router
23 verbose already installed skipping events@1.0.2 c:\Users\chouse\Desktop\react-router
24 verbose already installed skipping expect@^1.6.0 c:\Users\chouse\Desktop\react-router
25 verbose already installed skipping karma@^0.12.28 c:\Users\chouse\Desktop\react-router
26 verbose already installed skipping karma-chrome-launcher@^0.1.7 c:\Users\chouse\Desktop\react-router
27 verbose already installed skipping karma-cli@0.0.4 c:\Users\chouse\Desktop\react-router
28 verbose already installed skipping karma-firefox-launcher@^0.1.3 c:\Users\chouse\Desktop\react-router
29 verbose already installed skipping karma-mocha@^0.1.10 c:\Users\chouse\Desktop\react-router
30 verbose already installed skipping karma-sourcemap-loader@^0.3.2 c:\Users\chouse\Desktop\react-router
31 verbose already installed skipping karma-webpack@^1.3.1 c:\Users\chouse\Desktop\react-router
32 verbose already installed skipping marked@0.3.3 c:\Users\chouse\Desktop\react-router
33 verbose already installed skipping mocha@^2.0.1 c:\Users\chouse\Desktop\react-router
34 verbose already installed skipping pygmentize-bundled@^2.3.0 c:\Users\chouse\Desktop\react-router
35 verbose already installed skipping react@0.13.x c:\Users\chouse\Desktop\react-router
36 verbose already installed skipping rf-changelog@^0.4.0 c:\Users\chouse\Desktop\react-router
37 verbose already installed skipping rx@2.3.18 c:\Users\chouse\Desktop\react-router
38 verbose already installed skipping slash@^1.0.0 c:\Users\chouse\Desktop\react-router
39 verbose already installed skipping webpack@^1.4.13 c:\Users\chouse\Desktop\react-router
40 verbose already installed skipping webpack-dev-server@^1.6.6 c:\Users\chouse\Desktop\react-router
41 silly resolved []
42 info build c:\Users\chouse\Desktop\react-router
43 verbose linkStuff [ false, false, false, 'c:\\Users\\chouse\\Desktop' ]
44 info linkStuff react-router@1.0.0-beta3
45 verbose linkBins react-router@1.0.0-beta3
46 verbose linkMans react-router@1.0.0-beta3
47 verbose rebuildBundles react-router@1.0.0-beta3
48 verbose rebuildBundles [ '.bin',
48 verbose rebuildBundles   'babel',
48 verbose rebuildBundles   'babel-core',
48 verbose rebuildBundles   'babel-eslint',
48 verbose rebuildBundles   'babel-loader',
48 verbose rebuildBundles   'babel-plugin-object-assign',
48 verbose rebuildBundles   'bundle-loader',
48 verbose rebuildBundles   'eslint',
48 verbose rebuildBundles   'events',
48 verbose rebuildBundles   'expect',
48 verbose rebuildBundles   'invariant',
48 verbose rebuildBundles   'karma',
48 verbose rebuildBundles   'karma-chrome-launcher',
48 verbose rebuildBundles   'karma-cli',
48 verbose rebuildBundles   'karma-firefox-launcher',
48 verbose rebuildBundles   'karma-mocha',
48 verbose rebuildBundles   'karma-sourcemap-loader',
48 verbose rebuildBundles   'karma-webpack',
48 verbose rebuildBundles   'keymirror',
48 verbose rebuildBundles   'marked',
48 verbose rebuildBundles   'mocha',
48 verbose rebuildBundles   'node-libs-browser',
48 verbose rebuildBundles   'pygmentize-bundled',
48 verbose rebuildBundles   'qs',
48 verbose rebuildBundles   'react',
48 verbose rebuildBundles   'rf-changelog',
48 verbose rebuildBundles   'rx',
48 verbose rebuildBundles   'slash',
48 verbose rebuildBundles   'warning',
48 verbose rebuildBundles   'webpack',
48 verbose rebuildBundles   'webpack-dev-server' ]
49 verbose rebuild bundle c:\Users\chouse\Desktop\react-router\node_modules\node-libs-browser
50 info build c:\Users\chouse\Desktop\react-router\node_modules\node-libs-browser
51 info preinstall node-libs-browser@0.5.2
52 verbose linkStuff [ false,
52 verbose linkStuff   false,
52 verbose linkStuff   false,
52 verbose linkStuff   'c:\\Users\\chouse\\Desktop\\react-router\\node_modules' ]
53 info linkStuff node-libs-browser@0.5.2
54 verbose linkBins node-libs-browser@0.5.2
55 verbose linkMans node-libs-browser@0.5.2
56 verbose rebuildBundles node-libs-browser@0.5.2
57 verbose rebuildBundles [ 'assert',
57 verbose rebuildBundles   'browserify-zlib',
57 verbose rebuildBundles   'buffer',
57 verbose rebuildBundles   'console-browserify',
57 verbose rebuildBundles   'constants-browserify',
57 verbose rebuildBundles   'crypto-browserify',
57 verbose rebuildBundles   'domain-browser',
57 verbose rebuildBundles   'http-browserify',
57 verbose rebuildBundles   'https-browserify',
57 verbose rebuildBundles   'os-browserify',
57 verbose rebuildBundles   'path-browserify',
57 verbose rebuildBundles   'process',
57 verbose rebuildBundles   'punycode',
57 verbose rebuildBundles   'querystring-es3',
57 verbose rebuildBundles   'readable-stream',
57 verbose rebuildBundles   'stream-browserify',
57 verbose rebuildBundles   'string_decoder',
57 verbose rebuildBundles   'timers-browserify',
57 verbose rebuildBundles   'tty-browserify',
57 verbose rebuildBundles   'url',
57 verbose rebuildBundles   'util',
57 verbose rebuildBundles   'vm-browserify' ]
58 info install node-libs-browser@0.5.2
59 info postinstall node-libs-browser@0.5.2
60 info install react-router@1.0.0-beta3
61 info postinstall react-router@1.0.0-beta3
62 info prepublish react-router@1.0.0-beta3
63 verbose unsafe-perm in lifecycle true
64 info react-router@1.0.0-beta3 Failed to exec prepublish script
65 error react-router@1.0.0-beta3 prepublish: `npm run build`
65 error Exit status 1
66 error Failed at the react-router@1.0.0-beta3 prepublish script.
66 error This is most likely a problem with the react-router package,
66 error not with npm itself.
66 error Tell the author that this fails on your system:
66 error     npm run build
66 error You can get their info via:
66 error     npm owner ls react-router
66 error There is likely additional logging output above.
67 error System Windows_NT 6.1.7601
68 error command "c:\\Program Files (x86)\\nodejs\\node.exe" "c:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
 "--msvs_version=2013"
69 error cwd c:\Users\chouse\Desktop\react-router
70 error node -v v0.10.39
71 error npm -v 1.4.28
72 error code ELIFECYCLE
73 verbose exit [ 1, true ]

**Comment by mjackson** [comment]
Looks like the `prepublish` step is failing, most likely because we use a few [Bash scripts](https://github.com/rackt/react-router/tree/master/scripts) to create the build. Maybe we need a batch file?

**Comment by ryanflorence** [comment]
fixed by 4c6dc1b93484baec53b37fb4b9619006be7c5623 (unless its not, let us know)

**Comment by jure** [comment]
With this, I first got the error that 'babel' does not exist, because I didn't have it installed globally, and after fixing this, it fails due to "Unknown plugin 'object-assign'":

```
> babel ./modules -d lib --ignore '__tests__'

ReferenceError: Unknown plugin "object-assign"
    at PluginManager.subnormaliseString (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/transformation/file/plugin-manager.js:147:13)
    at PluginManager.add (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/transformation/file/plugin-manager.js:190:40)
    at File.buildTransformers (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/transformation/file/index.js:237:21)
    at new File (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/transformation/file/index.js:139:10)
    at Pipeline.transform (/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/transformation/pipeline.js:164:16)
    at Object.exports.transform (/usr/local/lib/node_modules/babel/lib/babel/util.js:40:22)
    at Object.exports.compile (/usr/local/lib/node_modules/babel/lib/babel/util.js:49:20)
    at write (/usr/local/lib/node_modules/babel/lib/babel/dir.js:19:21)
    at handleFile (/usr/local/lib/node_modules/babel/lib/babel/dir.js:40:7)
    at /usr/local/lib/node_modules/babel/lib/babel/dir.js:56:9

npm ERR! Darwin 14.5.0
npm ERR! argv "node" "/usr/local/bin/npm" "run" "build"
npm ERR! node v0.12.7
npm ERR! npm  v2.14.2
npm ERR! code ELIFECYCLE
npm ERR! react-router@1.0.0-rc1 build: `babel ./modules -d lib --ignore '__tests__'`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the react-router@1.0.0-rc1 build script 'babel ./modules -d lib --ignore '__tests__''.
npm ERR! This is most likely a problem with the react-router package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     babel ./modules -d lib --ignore '__tests__'
npm ERR! You can get their info via:
npm ERR!     npm owner ls react-router
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/src/react-webpack-node-2/node_modules/react-router/npm-debug.log
npm ERR! Darwin 14.5.0
npm ERR! argv "node" "/usr/local/bin/npm" "install"
npm ERR! node v0.12.7
npm ERR! npm  v2.14.2
npm ERR! code ELIFECYCLE

npm ERR! react-router@1.0.0-rc1 postinstall: `node -e "require('fs').stat('lib', function (e, s) { process.exit(e || !s.isDirectory() ? 1 : 0) })" || npm run build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the react-router@1.0.0-rc1 postinstall script 'node -e "require('fs').stat('lib', function (e, s) { process.exit(e || !s.isDirectory() ? 1 : 0) })" || npm run build'.
npm ERR! This is most likely a problem with the react-router package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node -e "require('fs').stat('lib', function (e, s) { process.exit(e || !s.isDirectory() ? 1 : 0) })" || npm run build
npm ERR! You can get their info via:
npm ERR!     npm owner ls react-router
npm ERR! There is likely additional logging output above.
```
